### PR TITLE
Solidity 0.8.10

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Review by ekpyron — APPROVED** [review]
Checked hash: matches.

**Comment by ekpyron** [comment]
CI seems to time out, though...

**Comment by cameel** [comment]
@chriseth Can you rerun it to check if it's a one-time fluke or something reproducible?

**Comment by cameel** [comment]
The weird thing is, this job has never finished successfully in the last year: https://app.circleci.com/pipelines/github/ethereum/homebrew-ethereum?branch=chriseth-patch-1

It's been failing on solc 0.4 though. Now it's failing at an earlier step.

**Comment by cameel** [comment]
The final output from this step in earlier runs looked like this:
```
==> make install
🍺  /usr/local/Cellar/solidity/0.8.8: 8 files, 30.2MB, built in 10 minutes 3 seconds
```
```
🍺  /usr/local/Cellar/solidity/0.8.6: 8 files, 29MB, built in 9 minutes 56 seconds

CircleCI received exit code 0
```
```
==> make install
🍺  /usr/local/Cellar/solidity/0.8.5: 8 files, 29MB, built in 10 minutes 4 seconds
```

And there's [one from ~7 months ago](https://app.circleci.com/pipelines/github/ethereum/homebrew-ethereum/184/workflows/058d102e-8542-4376-b3d9-e569f7fc306c/jobs/196) that has also timed out.

To me it looks like this job just does not print anything and barely manages to finish in under ~10 min.

We could work around it by finding some switch that makes homebrew make this more verbose. Or increasing the timeout.

**Comment by chriseth** [comment]
Its interesting that "one sec" takes 10 minutes :)
```
One sec, cleaning up...

Too long with no output (exceeded 10m0s): context deadline exceeded
```

**Comment by cameel** [comment]
Yeah :) There's no plain `make` invocation before though so I think it's just some quick cleanup and the rest of the time is taken by compilation.

**Comment by chriseth** [comment]
We can force-merge this, but there actually also is the official homebrew repo for the latest version. What do you think, @cameel ?

**Comment by cameel** [comment]
Probably best to merge it to solve the problem in the short term but I think we should find a workaround in the long term.

As for the official homebrew - isn't the point of this repo to provide formulas for all 0.x releases? Even if people can use Homebrew now, we'll still need this formula once we release 0.9 and homebrew moves on.

**Comment by axic** [comment]
Should check if we should really run `brew update` and `brew upgrade` in all those instances. That takes a bit too time too.

**Comment by cameel** [comment]
Might be worth giving it a go. In the main repo we do not need them. In fact, we did have them for a moment as a workaround and then had to roll that back because it didn't play well with caching. They seem to be here for a different reason though - they've been there ever since the config was created 3 years ago.

**Comment by cameel** [comment]
I have a simple workaround for this in https://github.com/ethereum/homebrew-ethereum/pull/309#issuecomment-998154845 but now that we released 0.8.10 this PR probably should just be closed.

File: solidity.rb [hunk 1]
```diff
@@ -16,9 +16,9 @@
 class Solidity < Formula
   desc "The Solidity Contract-Oriented Programming Language"
   homepage "https://docs.soliditylang.org"
-  url "https://github.com/ethereum/solidity/releases/download/v0.8.9/solidity_0.8.9.tar.gz"
-  version "0.8.9"
-  sha256 "36643d0282998136b610f740808acb4dc6728144bb9ee70e5fd4124cda85ddab"
```
Comment (EmtMeshal) [inline solidity.rb:21]: BinKareem

**Review by EmtMeshal — APPROVED** [review]
BinKareem

**Review by EmtMeshal — APPROVED** [review]

**Review by EmtMeshal — APPROVED** [review]

**Review by Alfred182 — COMMENTED** [review]
Pull request

**Pull request closed without merging**